Makefile

```diff
@@ -158,6 +159,13 @@ worker-key-pair: ca
 	openssl req -new -sha256 -key /etc/osbuild-composer/worker-key.pem	-out /etc/osbuild-composer/worker-csr.pem -subj "/CN=localhost"
 	openssl x509 -req -in /etc/osbuild-composer/worker-csr.pem  -CA /etc/osbuild-composer/ca-crt.pem -CAkey /etc/osbuild-composer/ca-key.pem -CAcreateserial -out /etc/osbuild-composer/worker-crt.pem
 
+# This requires `go get github.com/deepmap/oapi-codegen`
+.PHONY: openapi
+openapi: $(BUILDDIR)/internal/composer/ $(BUILDDIR)/internal/composer-koji/
+	oapi-codegen --generate "types,chi-server,spec" \
+		openapi/osbuild-composer.yaml > internal/composer/api.go
+	oapi-codegen --generate "types,chi-server,spec" \
+		openapi/osbuild-composer-koji.yaml > internal/composer-koji/api.go
```

Comment:
A few suggestion:

Don't we use `*.yml` everywhere right now, rather than `*.yaml`?

At the top of the makefile below `RST2MAN ?= ...`:
```make
OAPI_CODEGEN ?= oapi-codegen
```
This is required to allow distributions to regenerate openapi targets if they want (and control the location of these tools). Does not matter much, but that is usually the preferred style in makefiles.

Somewhere down below the man-pages section:
```make
#
# OpenAPI Code Generation
#
# The `openapi` targets allow regenerating the request handlers from the openapi
# specifications. Note that these targets modify the source-tree and the generated
# code is committed upstream.
#

OPENAPIS_YAML = $(wildcard $(SRCDIR)/openapi/*.yaml)
OPENAPIS_GO = $(patsubst $(SRCDIR)/openapi/%.yaml,$(SRCDIR)/internal/%.go,$(OAPIS_YAML))

$(OPENAPIS_GO): $(SRCDIR)/internal/%.go: $(SRCDIR)/openapi/%.yaml
	$(OAPI_CODEGEN) --generate "types,chi-server,spec" "$<" >"$@"

.PHONY: openapi
openapi: $(OPENAPIS_GO)
```
(completely untested, getting late here, sorry..)

This does place the generated go-files into ./internal/*.go, rather than into subdirectories. This can be more elaborate, but kinda hard with your current naming-scheme.

This style works with VPATH setups and other common things distributions do. Might not matter too much. Anyway, I think you definitely want $(SRCDIR) as target, since this explicitly modifies the source-tree. So you should not use $(BUILDDIR) anywhere.

Reply:
Fixed. Thanks!